README.md

```diff
@@ -80,6 +80,32 @@ Since 0.9.1:
 * `xml.validation.disallowDocTypeDecl`: Enable/disable if a fatal error is thrown if the incoming document contains a DOCTYPE declaration. Default is `false`.
 * `xml.validation.resolveExternalEntities`: Enable/disable resolve of external entities. Default is `false`.
         
+## Custom XML Extensions
+
+The [LSP4XML Language Server](https://github.com/angelozerr/lsp4xml) can be extended to support custom completion, hover, validation, rename, etc by using Java SPI feature. vscode-xml gives the capability to start the XML language server with your custom JAR which provides your custom XML support.
+
+To do that:
+
+ * create a Java project which provides custom XML extension which provides your custom completion, hover:
+    * create the XML extension like [MavenPlugin](https://github.com/angelozerr/lsp4xml-extensions-maven/blob/master/org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenPlugin.java).
+    * register in the XML extension your custom completion like [MavenCompletionParticipant](https://github.com/angelozerr/lsp4xml-extensions-maven/blob/master/org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenCompletionParticipant.java#L28)
```

Comment:
register your custom completion in the XML extension like

Reply:
@xorye updated